tutorial.md

```diff
@@ -1,3 +1,1555 @@
+
+
+
+
+    ''
+
+
+
+When we check the CAT tool for the Spanish vendor, we'll get "XTM". We did not provide any argument, but in the class it is set to default since that is the main CAT tool used in the particular agency. This can be changed in the actual code.
+
+
+```python
+VendorES.CatTool
+```
+
+
+
+
+    'XTM'
+
+
+
+The value for the argument Preferred can be True, False or None. This has an influence of the vendor's status.
+If Preferred is True, the vendor gets the status Preferred. If it is False, the vendor gets the status Back-up. If it is None, the vendor gets the status Potential.
+
+
+```python
+VendorFR.Preferred
+```
+
+
+
+
+    True
+
+
+
+
+```python
+VendorFR.Status
+```
+
+
+
+
+    'Preferred'
+
+
+
+
+```python
+VendorES.Preferred
+```
+
+
+
+
+    False
+
+
+
+
+```python
+VendorES.Status
+```
+
+
+
+
+    'Back-up'
+
+
+
+
+```python
+VendorNL.Preferred #Will not return anything since value is None
+```
+
+
+```python
+VendorNL.Status
+```
+
+
+
+
+    'Potential'
+
+
+
+### Functions
+The class also contains a couple of functions:
+- `SetVendorMail` to set an e-mail address for a specific vendor
+- `SetWordRate` to add a word rate in EUR/word for a specific vendor
+- `ChangeTool` to change the preferred CAT tool
+- `SetStatus` to change the vendor's status
+- `ToExcel` to write the data to an excel file
+- `ReadExcel` to print the data in the excel file to a dictionary, if the excel file exists
+- `ModExcel` to modify certain values for a vendor in the excel file, if the excel file exists
+
+We did not provide an e-mail for the Spanish vendor so we'll start with that:
+
+
+```python
+VendorES.SetVendorMail("e.dlbanda@outlook.com")
+VendorES.VendorMail
+```
+
+
+
+
+    'e.dlbanda@outlook.com'
+
+
+
+We did not set a word rate for the Dutch vendor, so now we can add one:
+
+
+```python
+VendorNL.SetWordRate(0.08)
+VendorNL.WordRate
+```
+
+
+
+
+    0.08
+
+
+
+Say, we did not yet know which CAT Tool the Spanish Vendor was going to use, but now we know that they will use Trados Studio. First it was set to 'XTM', we can change this as follows:
+
+
+```python
+VendorES.ChangeTool("Trados Studio")
+VendorES.CatTool
+```
+
+
+
+
+    'Trados Studio'
+
+
+
+Because we sat the value for preferred to True for the French vendor, they got the status "Preferred". If for some reason this changes, we can change the status easily, by changing the value for Preferred to False or None:
+
+
+```python
+VendorFR.SetStatus(False)
+VendorFR.Preferred
+```
+
+
+
+
+    False
+
+
+
+
+```python
+VendorFR.Status
+```
+
+
+
+
+    'Back-up'
+
+
+
+We can now write the data for the vendors to an excel file. Since every vendor has the value for the argument `ProjectName` and `SourceLang`, they will be written to the same excel file. We should execute this function for every vendor. Be careful because if you run this function multiple times for the same vendor, it will not overwrite but append and you will have duplicates in the excel file.
```

Comment:
The fact that you can generate duplicates is a weakness in the code.
You're doing a great job, so if you feel that you cannot fix this yet (you could, by checking for the existence of an entry before adding it), at least add it to the list of things that could/should be improved in the future.
You could also make this very explicit in the markdown:

```md
<div class="alert alert-danger">
If you run `.toExcel()` multiple times on the same vendor, it will generate multiple entries!
</div>
```

Which generates the following:
![image](https://github.com/montesmariana/intro_machine_learning_using_python/assets/54317904/2df7cebc-d1a5-46e6-a26e-f07abb5a4f32)